src/hyperlink/test/test_url.py

```diff
@@ -883,41 +883,41 @@ def check(param, expectation=defaultExpectation):
         check("port", "int or NoneType")
 
         with self.assertRaises(TypeError) as raised:
-            URL(path=[Unexpected()])
+            URL(path=[cast(Text, Unexpected())])
 
         assertRaised(raised, defaultExpectation, "path segment")
 
         with self.assertRaises(TypeError) as raised:
-            URL(query=[(u"name", Unexpected())])
+            URL(query=[(u"name", cast(Text, Unexpected()))])
 
         assertRaised(raised, defaultExpectation + " or NoneType",
                      "query parameter value")
 
         with self.assertRaises(TypeError) as raised:
-            URL(query=[(Unexpected(), u"value")])
+            URL(query=[(cast(Text, Unexpected()), u"value")])
 
         assertRaised(raised, defaultExpectation, "query parameter name")
         # No custom error message for this one, just want to make sure
         # non-2-tuples don't get through.
 
         with self.assertRaises(TypeError):
-            URL(query=[Unexpected()])
+            URL(query=[cast(Tuple[Text, Text], Unexpected())])
 
         with self.assertRaises(ValueError):
-            URL(query=[('k', 'v', 'vv')])
+            URL(query=[cast(Tuple[Text, Text], ('k', 'v', 'vv'))])
 
         with self.assertRaises(ValueError):
-            URL(query=[('k',)])
+            URL(query=[cast(Tuple[Text, Text], ('k',))])
 
         url = URL.from_text("https://valid.example.com/")
         with self.assertRaises(TypeError) as raised:
-            url.child(Unexpected())
+            url.child(cast(Text, Unexpected()))
         assertRaised(raised, defaultExpectation, "path segment")
         with self.assertRaises(TypeError) as raised:
-            url.sibling(Unexpected())
+            url.sibling(cast(Text, Unexpected()))
         assertRaised(raised, defaultExpectation, "path segment")
         with self.assertRaises(TypeError) as raised:
-            url.click(Unexpected())
+            url.click(cast(Text, Unexpected()))
```

Comment:
These casts are all because the tests are intentionally passing in incorrect types.